python/ray/tests/test_autoscaler.py

```diff
@@ -537,6 +537,106 @@ def _create_node(node_config, tags, count, _skip_wait=False):
         runner.assert_has_call("1.2.3.4", pattern=pattern_to_assert)
         return config
 
+    def testNodeTypeNameChange(self):
+        """
+        Tests that cluster launcher and autoscaler have correct behavior under
+        changes and deletions of node type keys.
+
+        Specifically if we change the key from "old-type" to "new-type", nodes
+        of type "old-type" are deleted and (if required by the config) replaced
+        by nodes of type "new-type".
+        """
+
+        # Default config with renamed node types, min_worker 1, docker off.
+        config = copy.deepcopy(MOCK_DEFAULT_CONFIG)
+        config["docker"] = {}
+        node_types = config["available_node_types"]
+        node_types["ray.head.old"] = node_types.pop("ray.head.default")
+        node_types["ray.worker.old"] = node_types.pop("ray.worker.default")
+        config["head_node_type"] = "ray.head.old"
+        node_types["ray.worker.old"]["min_workers"] = 1
+
+        # Create head and launch autoscaler
+        runner = MockProcessRunner()
+        self.provider = MockProvider()
+
+        config_path = self.write_config(config)
+        commands.get_or_create_head_node(
+            config,
+            printable_config_file=config_path,
+            no_restart=False,
+            restart_only=False,
+            yes=True,
+            override_cluster_name=None,
+            _provider=self.provider,
+            _runner=runner)
+        self.waitForNodes(1)
+        autoscaler = StandardAutoscaler(
+            config_path,
+            LoadMetrics(),
+            max_failures=0,
+            process_runner=runner,
+            update_interval_s=0)
+        autoscaler.update()
+        self.waitForNodes(2)
+        head = self.provider.non_terminated_nodes({
+            TAG_RAY_NODE_KIND: NODE_KIND_HEAD
+        })[0]
+        worker = self.provider.non_terminated_nodes({
+            TAG_RAY_NODE_KIND: NODE_KIND_WORKER
+        })[0]
```

Comment:
Could we assert that **all workers** have the correct types?

Reply:
Will make this clearer -- in this case there is only one worker.